generated/nifake/tests/test_session.py

```diff
@@ -246,10 +291,100 @@ def test_array_input_function(self):
+        boolean = True
+        int32 = 32
+        int64 = 64
+        enum = nifake.Turtle.LEONARDO
+        float = 1.23
+        float_enum = nifake.FloatEnum._6_5
+        array = [0, 1, 2]
+        array_size = len(array)
+        string = 'Testing is fun?'
+        self.side_effects_helper['ReturnMultipleTypes']['return'] = len(string)
+        self.side_effects_helper['ReturnMultipleTypes']['aBoolean'] = boolean
+        self.side_effects_helper['ReturnMultipleTypes']['anInt32'] = int32
+        self.side_effects_helper['ReturnMultipleTypes']['anInt64'] = int64
+        self.side_effects_helper['ReturnMultipleTypes']['anIntEnum'] = enum.value
+        self.side_effects_helper['ReturnMultipleTypes']['aFloat'] = float
+        self.side_effects_helper['ReturnMultipleTypes']['aFloatEnum'] = float_enum.value
+        self.side_effects_helper['ReturnMultipleTypes']['anArray'] = array
+        self.side_effects_helper['ReturnMultipleTypes']['aString'] = string
+        self.side_effects_helper['ReturnMultipleTypes']['return'] = 0
+        with nifake.Session('dev1') as session:
+            result_boolean, result_int32, result_int64, result_enum, result_float, result_float_enum, result_array, result_string = session.return_multiple_types(array_size)
+            assert result_boolean == boolean
+            assert result_int32 == int32
+            assert result_int64 == int64
+            assert result_enum == enum
+            assert result_float == float
+            assert result_float_enum == float_enum
+            assert result_array == array
+            assert result_string == string
+            assert self.patched_library.niFake_ReturnMultipleTypes.call_count == 2
+
+    def test_multiple_array_types(self):
+        self.patched_library.niFake_MultipleArrayTypes.side_effect = self.side_effects_helper.niFake_MultipleArrayTypes
+        passed_in_array = [0.0, 1.0]
+        passed_in_array_size = len(passed_in_array)
+        fixed_size_array = [2.0, 3.0, 4.0]
+        len_array = [5.0, 6.0, 7.0, 8.0]
+        self.side_effects_helper['MultipleArrayTypes']['passedInArray'] = passed_in_array
+        self.side_effects_helper['MultipleArrayTypes']['aFixedArray'] = fixed_size_array
+        self.side_effects_helper['MultipleArrayTypes']['return'] = 0
+        with nifake.Session('dev1') as session:
+            passed_in_array_result, fixed_size_array_result = session.multiple_array_types(passed_in_array_size, len_array)
+            assert passed_in_array == passed_in_array_result
+            assert fixed_size_array == fixed_size_array_result
+
+    def test_parameters_are_multiple_types(self):
+        self.patched_library.niFake_ParametersAreMultipleTypes.side_effect = self.side_effects_helper.niFake_ParametersAreMultipleTypes
+        boolean = True
+        int32 = 32
+        int64 = 64
+        enum = nifake.Turtle.LEONARDO
+        float = 1.23
```

Comment:
Please don't use the type name as the variable name. Just add _val to it or something `enum_val`, `float_val`, etc.

Reply:
Fixed